src/ui/atoms/typography.svelte

```diff
@@ -0,0 +1,132 @@
+<script module>
+  import { type Snippet } from "svelte";
+
+  type Size =
+    | "heading-2xl"
+    | "heading-xl"
+    | "heading-lg"
+    | "heading-md"
+    | "body-base"
+    | "body-small"
+    | "label-button"
+    | "label-default"
+    | "caption-default"
+    | "caption-link";
+
+  type Props = {
+    size?: Size;
+    children?: Snippet;
+  };
+</script>
+
+<script lang="ts">
+  const { children, size = "body-base" }: Props = $props();
+</script>
+
+<span class="rcb-typography rcb-typography-{size}">
+  {@render children?.()}
+</span>
+
+<style>
+  /* svelte-ignore css-unused-selector */
+  .rcb-typography-heading-2xl {
+    font: var(--rc-text-heading2xl-mobile);
+  }
```

Comment:
I feel like we should just use `style` directly instead of having to define all these css classes

Reply:
sounds good 👍🏻

Reply:
The change got a little tricky having to convert the namings "-heading-2xl-" -> "headingXl" and also with media queries in the middle. Since it's a step forward and we will make bigger strides soon with the design system I will keep it as it is.